qsearchfield_mac.mm

```diff
@@ -129,32 +129,39 @@ -(BOOL)performKeyEquivalent:(NSEvent*)event;
 
 @implementation QocoaSearchField
 -(BOOL)performKeyEquivalent:(NSEvent*)event {
-    if ([event type] == NSKeyDown && [event modifierFlags] & NSCommandKeyMask)
-    {
-        QString keyString = toQString([event characters]);
-        if (keyString == "a")  // Cmd+a
-        {
-            [self performSelector:@selector(selectText:)];
-            return YES;
-        }
-        else if (keyString == "c")  // Cmd+c
-        {
-            QClipboard* clipboard = QApplication::clipboard();
-            clipboard->setText(toQString([self stringValue]));
-            return YES;
-        }
-        else if (keyString == "v")  // Cmd+v
-        {
-            QClipboard* clipboard = QApplication::clipboard();
-            [self setStringValue:fromQString(clipboard->text())];
-            return YES;
-        }
-        else if (keyString == "x")  // Cmd+x
+    // First, check if we have the focus.
+    // If no, it probably means this event isn't for us.
+    NSResponder* firstResponder = [[NSApp keyWindow] firstResponder];
+    if ([firstResponder isKindOfClass:[NSText class]] &&
+            [(NSText*)firstResponder delegate] == self) {
```

Comment:
Don't get confused: only those 3 lines matter.
All other modifications are just indentation changes because of this new "if" block.
This is how I did in Clementine [1] but actually it might be better to this check but return, to avoid modifying too many lines (as among other things, this makes git blame less obvious).
Let me know if you want to change this or if it is OK like this.

[1]: https://github.com/clementine-player/Clementine/commit/fe7bc917deb7c0512baddbe9b0f9e4d9c5b69a6c#diff-38bb1f57126fbec73a6437f7e592221b